src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,190 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final String INDEX_OUT_OF_BOUNDS = "Index is out of bounds of list";
+    private ListNode<T> head;
+    private ListNode<T> tail;
+    private int size;
+
+    private class ListNode<T> {
+        private T value;
+        private ListNode prev;
+        private ListNode next;
+
+        ListNode(T value) {
+            this.value = value;
+        }
+    }
+
     @Override
     public void add(T value) {
+        if (isEmpty()) {
+            head = new ListNode<>(value);
+            tail = head;
+            size = 1;
+        } else {
+            tail.next = new ListNode<>(value);
+            tail.next.prev = tail;
+            tail = tail.next;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        ListNode<T> node = getListNode(index);
+        ListNode<T> newNode = new ListNode<>(value);
+        boolean isAfterTailInsert = false;
```

Comment:
Try to refactor this method `add(T value, int index)` without this variable `isAfterTailInsert`

You can use this example:
```
    @Override
    public void add(T value, int index) {
        checkIndex(index);
        if(index == size) {
            add(value);
        } else {
            Node<T> currentNode = getCurrentNodeBy(index);
            Node<T> newNode = new Node<>(currentNode.previous, value, currentNode);

            if (head == currentNode) {
                head = newNode;
            } else {
                currentNode.previous.next = newNode;
            }
            currentNode.previous = newNode;
            size++;
        }
    }
```